File: csharp/etl/Etl.cs
using System;
using System.Collections.Generic;

public static class Etl
{
    public static IDictionary<string, int> Transform(IDictionary<int, IList<string>> old)
    {

        IDictionary<string, int> res = new Dictionary<string, int>();
        foreach (int val in old.Keys)
        {
            foreach(string key in old[val])
            {
                res.Add(key.ToLowerInvariant(), val);
            }
        }
        return res;
    }
}

File: kotlin/etl/src/main/kotlin/etl.kt
object ETL {
    fun transform(map: Map<Int, List<Char>>): Map<Char, Int> {
        var newMap: Map<Char, Int> = mutableMapOf()
        for ((k, v) in map.iterator()) {
            for (e in v.iterator()) {
                newMap += e.toLowerCase() to k
            }
        }
        return newMap
    }
}